### DISCUSSION TITLE
Unsupported MJPEG/YUV format after update from 17.0.0.0 to 18.0.0.0

### DISCUSSION THREAD

**Discussion by BassXX** [awawa-dev/HyperHDR#323] [Q&A]
Hi,

just updated my HyperHDR and now I get tons of these Logs:

[V4L2:USB3.0 HD VIDE] Unsupported MJPEG/YUV format. Please contact HyperHDR developers! (info: UNSUPPORTED YUV DECODER: 2)

What can I do to fix that? 
Thanks & Regards

**Comment by awawa-dev** [comment]
Hi
In case of HDR your grabber is providing non-standard MJPEG YUV internal format (at least none of my own grabbers use it I think, so can't implement it) [TJSAMP_420](https://jpeg-turbo.dpldocs.info/libjpeg.turbojpeg.TJSAMP.html#TJSAMP_420) instead of supported [TJSAMP_422](https://jpeg-turbo.dpldocs.info/libjpeg.turbojpeg.TJSAMP.html#TJSAMP_422). Which model of grabber do you have?

**Reply by BassXX** [reply]
The name is ZasLuke Capture card. Probably a no-name brand. Does It mean I have to buy another grabber? 

it’s interesting, because it was working somehow with v17

**Reply by awawa-dev** [reply]
Yes it was working in v17. Probably till the patch introduced long time ago, even before v18beta2 (https://github.com/awawa-dev/HyperHDR/commit/75f8c3347b63ceefcdafb477172245f8685bc81e) changed the situation.  As you can see the reward is quite significant: 30% of performance and much lower CPU usage. If I had known about it then, maybe I would have thought of a workaround. The incompatibility for such unusual grabbers is limited only for HDR, SDR should still work. If you are interested then I may think about some work-around but I need a tester or a hardware.

**Reply by BassXX** [reply]
I see. But I seem to get the problem also for non-HDR devices. Do I need to disable SDRtoHDR Mapping or something?

Sure I am willing to test :) if it turns out to be too complicated, I could also get another supported  grabber

**Reply by awawa-dev** [reply]
Yes, disabling HDR tone mapping in the grabber's properties also forces HyperHDR to use alternative libturbojpeg MJPEG to RGB conversion (the old way from v17). Probably the workaround is to handle such grabber in old way also for HDR, without optimization and against v18 advertised 30% gains ;) Will take a look next weekend.

**Reply by BassXX** [reply]
Just checked - SDR works fine.

thanks an keep me updated :)

**Comment by awawa-dev** [comment]
Please test a build from Github Action https://github.com/awawa-dev/HyperHDR/actions/runs/2989926680 when it's finished. There will be installers in the artifacts section. 
I think that [TJSAMP_420](https://jpeg-turbo.dpldocs.info/libjpeg.turbojpeg.TJSAMP.html#TJSAMP_420) is the same codec as I420 that I implemented long time ago in HyperHDR but I'm not sure. If you test it, please let me know the result.

**Reply by BassXX** [reply]
Hi there! Thanks for your answer and your work on a testbuild.

I can test, but not before end of septmeber unfortunately. Thanks for your patience

**Reply by sorcererlc** [reply]
Hi

I installed the test build and I can confirm that my Zasluke capture card is working properly with HDR to SDR tone mapping enabled.

Thanks!

**Reply by awawa-dev** [reply]
Thank you for the feedback. So the patch will be merge soon with the upstream.
